external-api/api-tests/v1/eStore.api-test.ts

```diff
@@ -136,20 +141,29 @@ describe('/estore', () => {
       await post(payload).to('/estore');
 
       // Look up for the deal ID in the collection
-      expect(mockFindOne).toHaveBeenCalled();
+      expect(mockFindOne).toHaveBeenCalledWith({ 'payload.dealId': { $eq: new ObjectId(payload.dealId) } });
 
       // Insert a new entry in the collection
-      expect(mockInsertOne).toHaveBeenCalled();
+      expect(mockInsertOne).toHaveBeenCalledWith({
+        payload,
+        timestamp: getNowAsEpoch,
```

Comment:
Think this should be `getNowAsEpoch()` .

That might make the test a bit flakey though so `expect.any(Number)'` might be better